test/tools/cmap_spec_runner.ts

```diff
@@ -0,0 +1,336 @@
+import { expect } from 'chai';
+import { EventEmitter } from 'events';
+import * as util from 'util';
+
+import { Connection } from '../../src';
+import { ConnectionPool } from '../../src/cmap/connection_pool';
+import { sleep } from './utils';
+
+type cmapOperation =
+  | { name: 'start' | 'waitForThread'; target: string }
+  | { name: 'wait'; ms: number }
+  | { name: 'waitForEvent'; event: string; count: number }
+  | { name: 'checkOut'; thread: string; label: string }
+  | { name: 'checkIn'; connection: string }
+  | { name: 'clear' | 'close' | 'ready' };
+
+type cmapPoolOptions = {
+  maxPoolSize?: number;
+  minPoolSize?: number;
+  maxIdleTimeMS?: number;
+  waitQueueTimeoutMS?: number;
+};
+
+type cmapEvent = {
+  type: string;
+  address?: 42;
```

Comment:
Per the other 42 comment, these have to be `42 | '42'` Should we pull it out to a named type to indicate it's magical purpose?

```ts
type TheMeaningOfLife = 42 | '42';
```

Reply:
I think the link to the CMAP spec readme heading is sufficient, and I have the same response here as on the other comment - let's have the bare minimum logic that gets the test working for the existing definitions. You may notice that the thread is supposed to be able to be specified on any command, but I excluded it from types for any where it actually wouldn't be defined. I want the runner to be representative of actual expected input that we have actually seen. Everything else will error if new things are added, and can be addressed then.